File: csharp/sprint5/D/Solution.cs
#if REMOTE_JUDGE
#else
public class Node
{
    public int Value { get; set; }
    public Node Left { get; set; }
    public Node Right { get; set; }

    public Node(int value)
    {
        Value = value;
        Left = null;
        Right = null;
    }
}
#endif

public class Solution
{
    public static bool Solve(Node left, Node right)
    {
        // Your code
        // “ヽ(´▽｀)ノ”
    }

    private static void Test()
    {
        var node1 = new Node(1);
        var node2 = new Node(2);
        var node3 = new Node(3)
        {
            Left = node1,
            Right = node2
        };

        var node4 = new Node(1);
        var node5 = new Node(2);
        var node6 = new Node(3)
        {
            Left = node4,
            Right = node5
        };

        Console.WriteLine(Solve(node3, node6));
    }
}

File: kotlin/sprint5/D/Solution.kt
// <template>
class Node {
    var value: Int
    var left: Node?
    var right: Node?

    constructor(value: Int) {
        this.value = value
        right = null
        left = null
    }

    constructor(value: Int, left: Node?, right: Node?) {
        this.value = value
        this.left = left
        this.right = right
    }
}
// <template>


fun treeSolution(head1: Node?, head2: Node?): Boolean {
    // Your code
    // “ヽ(´▽｀)ノ”
}


fun test() {
    val node1 = Node(1, null, null)
    val node2 = Node(2, null, null)
    val node3 = Node(3, node1, node2)
    val node4 = Node(1, null, null)
    val node5 = Node(2, null, null)
    val node6 = Node(3, node4, node5)
    assert(treeSolution(node3, node6))
}
